src/js/jsx/sections/layers/LayersPanel.jsx

```diff
@@ -467,7 +467,7 @@ define(function (require, exports, module) {
 
                 var dragSource = collection.pluck(this.state.dragTargets, "id");
 
-                flux.actions.layers.reorder(doc, dragSource, dropIndex)
+                return flux.actions.layers.reorder(doc, dragSource, dropIndex)
```

Comment:
The reason behind this is that the "dragging" state should maintain until the drop process is finished. Without this, the dropped layer will snap back after dropping, or you have to add information into the draganddrop store to represents a `dropping` state. I think the promise way is a simpler and does not affect user experience.  I also added the reason here https://github.com/adobe-photoshop/spaces-design/pull/1829/files#diff-9adf1065d9513443f0903ee1061b1b0dR182